README.md

```diff
@@ -20,212 +20,324 @@ To install shelf-lib-js for development, run the following commands:
 
 
-Reference Creation
-------------------
+### `shelfLib~initReference(refName, authToken)`
 
-    // Grab a reference to a bucket.
-    var reference = shelfLib.initReference("refName", "<super secret pyshelf API key>");
+* `refName` `{string}` - Could also be called "bucket", "storage" or "shelf". This represents a specific storage space.
+* `authToken` `{string}` - The authentication token used to authenticate with the `reference`
+* Returns: `{Object}` - A [reference](#reference) object.
 
+Example:
 
-Artifact Reference Creation
----------------------------
-
-    // Grab the artifact reference at location "/path".
-    var artifact = reference.initArtifact("path");
+    var reference = shelfLib.initReference("shadow", "bDMKbnnzrDlmNs34NHGkvHOWDVdE4okV");
 
 
-Uploading
+reference
 ---------
 
-To upload an artifact with the contents from a variable:
+Represents what people might call a "reference name", "bucket", "storage" or "shelf". See [the Shelf documentation](https://github.com/not-nexus/shelf/blob/master/docs/configuration.md) for more information.
 
-    artifact.upload("Hello world!").then((uploadLocation) => {
-        console.log("Uploaded content to: " + uploadLocation);
-    }, (err) => {
-        console.log("Hit an error: " + err);
-    });
 
-To upload an artifact from a file:
+### `reference~initArtifact(path)`
 
-    artifact.uploadFromFile("./file-to-upload.txt").then((uploadLocation) => {
-        console.log("Uploaded content to: " + uploadLocation);
-    }, (err) => {
-        console.log("Hit an error: " + err);
-    });
+* `path` `{string}` - Just the path to the artifact inside of the reference. It is `<path>` in `https://api.shelf.com/shadow/artifact/<path>`.
+* Returns: `{Object}` - An [artifact](#artifact) object.
+
+Example:
 
-Or:
+    var artifact = reference.initArtifact("/my/path");
 
-    var fileReadStream = fs.createReadStream("./file-to-upload.txt");
 
-    artifact.uploadFromFile(fileReadStream).then((uploadLocation) => {
-        console.log("Uploaded content to: " + uploadLocation);
-    }, (err) => {
-        console.log("Hit an error: " + err);
-    });
+### `reference~initArtifactWithTimestamp(path)`
 
+* `path` `{string}` - Just the path to the artifact inside of the reference. It is `<path>` in `https://api.shelf.com/shadow/artifact/<path>/2016-11-01T14:37:04.151Z`.
+* Returns: `{Object}`
 
-Downloading
------------
+Example:
 
-> WARNING: Do not use `shelfLib~Artifact.download` for large artifacts. By default node's max memory size is 1.76GB (for 64 bit systems). This will blow up if your artifact is approaching that size. Use `shelfLib~Artifact.downloadToFile` instead.
+    var artifact = reference.initWithTimestampArtifact("/my/path");
 
-To download the contents and receive it in a variable:
 
-    artifact.download().then((data) => {
-        console.log("Got data: " + data);
-    }, (err) => {
-        console.log("Hit an error: " + err);
-    });
+### `reference.initSearch([path])`
```

Comment:
This one uses a dot. Others use a tilde. I think, if we are mirroring Node, we'd use a dot in all of the headings.